### DISCUSSION TITLE
Upload to storage with the REST API and a base64 file string

### DISCUSSION THREAD

**Discussion by katewhite** [supabase/supabase#22582] [Questions]
I'm able to upload to my storage bucket using Postman and uploading a file with a binary body, uploaded directly to Postman. The cURL request looks like this as a result:
```
curl --location 'https://<myprojectid>.supabase.co/storage/v1/object/mybucket/test.png' \
--header 'Content-Type: image/png' \
--header 'apikey: <myapikey>' \
--header 'Authorization: Bearer <mytoken>' \
--data '@/Users/kate/Desktop/test.png'
```

But I haven't been able to get this same call to work with a base64 string for the file. I've tried changing `--data` to `--data-binary` and pasting the full string. It uploads properly, but the file is corrupt or otherwise incomplete/nonfunctional in the bucket.

Can anyone help me figure out how to accomplish this with the raw data string rather than a relative filepath (the filestring is what the upload mechanism in Draftbit provides, so I have no relative path to the file).

**Comment by GaryAustin1** [comment]
Not a file format expert, but I'll just throw out there I've seen users try and use a base64 format as img data and don't believe that works.  Base64 is normally used in the actual url string for an img tag.   Just a hint at something to investigate.
